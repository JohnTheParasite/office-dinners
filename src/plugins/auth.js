import AuthService from "@/services/authService";

export const Auth = {
  install(Vue) {
    Vue.prototype.$authService = AuthService;
  },
};