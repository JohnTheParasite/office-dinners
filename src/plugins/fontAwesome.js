import { library } from "@fortawesome/fontawesome-svg-core"
import {
  faArrowRightToBracket,
  faArrowUpRightFromSquare,
  faBan,
  faBurger,
  faChartLine,
  faCheck,
  faChevronDown,
  faChevronLeft,
  faChevronRight,
  faEye,
  faEyeSlash,
  faHandHoldingDollar,
  faHeart,
  faPencil,
  faPercent,
  faTrashCan,
  faUser,
  faUsers,
  faUtensils,
  faXmark
} from "@fortawesome/free-solid-svg-icons"
import { FontAwesomeIcon } from "@fortawesome/vue-fontawesome"
import Vue from "vue"

library.add(faChevronLeft)
library.add(faChevronRight)
library.add(faChevronDown)
library.add(faArrowUpRightFromSquare)
library.add(faCheck)
library.add(faPencil)
library.add(faTrashCan)
library.add(faBan)
library.add(faPercent)
library.add(faChartLine)
library.add(faUsers)
library.add(faBurger)
library.add(faHandHoldingDollar)
library.add(faUtensils)
library.add(faHeart)
library.add(faUser)
library.add(faArrowRightToBracket)
library.add(faXmark)
library.add(faEye)
library.add(faEyeSlash)
Vue.component("font-awesome-icon", FontAwesomeIcon)
