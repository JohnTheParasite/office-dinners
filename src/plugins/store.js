import store from "@/store/store"

export const Store = {
  install(Vue) {
    Vue.prototype.$store = store
  }
}
