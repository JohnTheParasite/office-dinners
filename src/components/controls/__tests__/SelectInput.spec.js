import { createLocalVue, mount, shallowMount } from "@vue/test-utils"
import selectInput from "@/components/controls/SelectInput"
import i18n from "@/i18n"
import "@/plugins/fontAwesome"

describe("selectInput", () => {
  const localVue = new createLocalVue()
  localVue.use(i18n)

  let mockOptions = [
    { text: "first element", value: 1, selected: false },
    { text: "second element", value: 2, selected: false }
  ]

  it("selectInput should render", () => {
    let selectInputElement = shallowMount(selectInput, { localVue, i18n, propsData: { options: mockOptions } })
    expect(selectInputElement).toMatchSnapshot()
  })

  it("clearInput() should work", () => {
    let selectInputElement = shallowMount(selectInput, { localVue, i18n, propsData: { options: mockOptions } })
    selectInputElement.vm.$refs.input.value = "testValue"
    expect(selectInputElement.vm.$refs.input.value).toBe("testValue")
    selectInputElement.vm.clearInput()
    expect(selectInputElement.vm.$refs.input.value).toBe("")
  })

  it("setValue() should work", () => {
    let selectInputElement = shallowMount(selectInput, { localVue, i18n, propsData: { options: mockOptions } })
    expect(selectInputElement.vm.filter).toBe("")
    selectInputElement.vm.setValue("testValue")
    expect(selectInputElement.vm.filter).toBe("testValue")
  })

  it("selectOption(option) should work", () => {
    let selectInputElement = shallowMount(selectInput, { localVue, i18n, propsData: { options: mockOptions } })
    expect(selectInputElement.vm.value).toBe(1)
    selectInputElement.vm.selectOption(mockOptions[1])
    expect(selectInputElement.vm.value).toBe(2)
  })

  it("enableOpenable() should work", () => {
    let selectInputElement = shallowMount(selectInput, { localVue, i18n, propsData: { options: mockOptions } })
    expect(selectInputElement.vm.showOpenable).toBeFalsy()
    selectInputElement.vm.enableOpenable()
    expect(selectInputElement.vm.showOpenable).toBeTruthy()
  })

  it("toggleOpenable() should work", () => {
    let selectInputElement = shallowMount(selectInput, { localVue, i18n, propsData: { options: mockOptions } })
    expect(selectInputElement.vm.showOpenable).toBeFalsy()
    selectInputElement.vm.toggleOpenable()
    expect(selectInputElement.vm.showOpenable).toBeTruthy()
  })

  it("should onDocumentClick(e) function work", () => {
    const component = {
      template: '<div><select-input :options="options"></select-input></div>',
      components: { selectInput },
      data() {
        return {
          options: mockOptions
        }
      }
    }

    let divElement = mount(component, { localVue, i18n })
    let selectInputElement = divElement.findComponent(selectInput)

    expect(selectInputElement.vm.showOpenable).toBeFalsy()
    selectInputElement.vm.enableOpenable()
    expect(selectInputElement.vm.showOpenable).toBeTruthy()
    selectInputElement.vm.onDocumentClick({ target: selectInputElement.vm.$el })
    expect(selectInputElement.vm.showOpenable).toBeTruthy()

    selectInputElement.vm.onDocumentClick({ target: divElement.vm.$el })
    expect(selectInputElement.vm.showOpenable).toBeFalsy()
  })
})
