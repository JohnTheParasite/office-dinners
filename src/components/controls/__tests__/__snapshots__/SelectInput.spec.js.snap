// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`selectInput selectInput should render 1`] = `
<div class="openable-container">
  <div class="input-select">
    <text-input-stub label="" description="" name="" type="text" placeholder="first element" warningmessage="errors.fieldIsRequired" icon="" oniconclick="[Function]"></text-input-stub>
    <div class="square">
      <font-awesome-icon-stub icon="fa-solid fa-chevron-down"></font-awesome-icon-stub>
    </div>
  </div>
  <div class="openable-options">
    <div class="option selected">
      <div class="text">first element</div>
      <font-awesome-icon-stub icon="fa-solid fa-check"></font-awesome-icon-stub>
    </div>
    <div class="option">
      <div class="text">second element</div>
      <!---->
    </div>
    <!---->
  </div>
</div>
`;
